subsys/task_wdt/task_wdt.c

```diff
@@ -77,9 +77,14 @@ static void schedule_next_timeout(int64_t current_ticks)
 		}
 	}
 
-	/* update task wdt kernel timer */
-	k_timer_user_data_set(&timer, (void *)next_channel_id);
-	k_timer_start(&timer, K_TIMEOUT_ABS_TICKS(next_timeout), K_FOREVER);
+	if (!IS_ENABLED(CONFIG_TASK_WDT_HW_FALLBACK) &&
+		next_timeout == INT64_MAX) {
+		k_timer_stop(&timer);
+	} else {
+		/* update task wdt kernel timer */
+		k_timer_user_data_set(&timer, (void *)next_channel_id);
+		k_timer_start(&timer, K_TIMEOUT_ABS_TICKS(next_timeout), K_FOREVER);
+	}
```

Comment:
Would this not be sufficient?

```suggestion
	if (next_timeout != INT64_MAX) {
		/* update task wdt kernel timer */
		k_timer_user_data_set(&timer, (void *)next_channel_id);
		k_timer_start(&timer, K_TIMEOUT_ABS_TICKS(next_timeout), K_FOREVER);
	}
```

Reply:
Yes, that would also be ok.

`(!IS_ENABLED(CONFIG_TASK_WDT_HW_FALLBACK)` check is basically micro-optimization, as compiler can optimize that block out when CONFIG_TASK_WDT_HW_FALLBACK=y. But that doesn't make huge difference.